packages/webpack-pwa-manifest-plugin/tests/runTest.js

```diff
@@ -0,0 +1,67 @@
+const webpack = require('webpack');
+const path = require('path');
+const ls = require('list-directory-contents');
+const fs = require('fs');
+const assert = require('assert');
+const chalk = require('chalk');
+
+function run(name, next) {
+  console.time(name);
+  if (!name) {
+    return console.log('End.');
+  }
+  require(path.resolve(`./tests/${name}/build/webpack.config.js`))().then(config => {
+    console.log(`"${name}": building...`);
+    webpack(config, (err, stats) => {
+      if (err) throw err;
+      console.log(`"${name}": testing...`);
+      const testPath = path.join(__dirname, name, 'test');
+      const testPathLength = testPath.length;
+      // Get the file contents for the directory we want to match
+      ls(testPath, (err, tree) => {
+        if (err) throw err;
+
+        const testTree = tree.filter(i => /.*\..+/.test(i));
+        const testContent = testTree.map(i => i.substring(testPathLength));
+        const testContentLength = testContent.length;
+        const outputPath = path.join(__dirname, name, 'output');
+        const outputPathLength = outputPath.length;
+
+        // Get the file contents for the directory we just generated
+        ls(outputPath, (err, outputTree) => {
+          if (err) throw err;
+
+          const outputContent = outputTree.filter(i => /.*\..+/.test(i));
+          const outputContentLength = outputContent.length;
+          if (testContentLength === outputContentLength) {
+            const result = outputContent.reduce((previous, current) => {
```

Comment:
`reduce` obfuscates the intent more than alternatives -- either write out the for-loop or run `filter` and get the `.length`